flepimop/gempyor_pkg/src/gempyor/simulate.py

```diff
@@ -156,126 +156,20 @@
+from .shared_cli import config_files_argument, config_file_options, parse_config_files, cli, click_helpstring
 
 # from .profile import profile_options
 
-
-@click.command()
-@click.option(
-    "-c",
-    "--config",
-    "config_filepath",
-    envvar=["CONFIG_PATH", "CONFIG_PATH"],
-    type=click.Path(exists=True),
-    required=True,
-    help="configuration file for this simulation",
-)
-@click.option(
-    "-s",
-    "--seir_modifiers_scenario",
-    "seir_modifiers_scenarios",
-    envvar="FLEPI_SEIR_SCENARIO",
-    type=str,
-    default=[],
-    multiple=True,
-    help="override the NPI scenario(s) run for this simulation [supports multiple NPI scenarios: `-s Wuhan -s None`]",
-)
-@click.option(
-    "-d",
-    "--outcome_modifiers_scenario",
-    "outcome_modifiers_scenarios",
-    envvar="FLEPI_OUTCOME_SCENARIO",
-    type=str,
-    default=[],
-    multiple=True,
-    help="Scenario of outcomes to run",
-)
-@click.option(
-    "-n",
-    "--nslots",
-    envvar="FLEPI_NUM_SLOTS",
-    type=click.IntRange(min=1),
-    help="override the # of simulation runs in the config file",
-)
-@click.option(
-    "-i",
-    "--first_sim_index",
-    envvar="FIRST_SIM_INDEX",
-    type=click.IntRange(min=1),
-    default=1,
-    show_default=True,
-    help="The index of the first simulation",
-)
-@click.option(
-    "-j",
-    "--jobs",
-    envvar="FLEPI_NJOBS",
-    type=click.IntRange(min=1),
-    default=multiprocessing.cpu_count(),
-    show_default=True,
-    help="the parallelization factor",
-)
-@click.option(
-    "--stochastic/--non-stochastic",
-    "--stochastic/--non-stochastic",
-    "stoch_traj_flag",
-    envvar="FLEPI_STOCHASTIC_RUN",
-    type=bool,
-    default=False,
-    help="Flag determining whether to run stochastic simulations or not",
-)
-@click.option(
-    "--in-id",
-    "--in-id",
-    "in_run_id",
-    envvar="FLEPI_RUN_INDEX",
-    type=str,
-    default=file_paths.run_id(),
-    show_default=True,
-    help="Unique identifier for the run",
-)  # Default does not make sense here
-@click.option(
-    "--out-id",
-    "--out-id",
-    "out_run_id",
-    envvar="FLEPI_RUN_INDEX",
-    type=str,
-    default=None,
-    show_default=True,
-    help="Unique identifier for the run",
-)
-@click.option(
-    "--in-prefix",
-    "--in-prefix",
-    "in_prefix",
-    envvar="FLEPI_PREFIX",
-    type=str,
-    default=None,
-    show_default=True,
-    help="unique identifier for the run",
-)
-@click.option(
-    "--write-csv/--no-write-csv",
-    default=False,
-    show_default=True,
-    help="write CSV output at end of simulation",
-)
-@click.option(
-    "--write-parquet/--no-write-parquet",
-    default=True,
-    show_default=True,
-    help="write parquet file output at end of simulation",
-)
 # @profile_options
 # @profile()
+@cli.command(params=[config_files_argument] + list(config_file_options.values()))
+@click_helpstring([config_files_argument] + list(config_file_options.values()))
 def simulate(
```

Comment:
This seems to have expanded the `--help` output quite a bit. From 
```
twillard@epid-iss-MacBook-Pro ~/D/G/H/f/f/gempyor_pkg (main) [1]> gempyor-simulate --help                                                                 (/Users/twillard/Desktop/GitHub/HopkinsIDD/flepiMoP/venv)
Usage: gempyor-simulate [OPTIONS]

Options:
  -c, --config PATH               configuration file for this simulation
                                  [required]
  -s, --seir_modifiers_scenario TEXT
                                  override the NPI scenario(s) run for this
                                  simulation [supports multiple NPI scenarios:
                                  `-s Wuhan -s None`]
  -d, --outcome_modifiers_scenario TEXT
                                  Scenario of outcomes to run
  -n, --nslots INTEGER RANGE      override the # of simulation runs in the
                                  config file  [x>=1]
  -i, --first_sim_index INTEGER RANGE
                                  The index of the first simulation  [default:
                                  1; x>=1]
  -j, --jobs INTEGER RANGE        the parallelization factor  [default: 14;
                                  x>=1]
  --stochastic, --stochastic / --non-stochastic, --non-stochastic
                                  Flag determining whether to run stochastic
                                  simulations or not
  --in-id, --in-id TEXT           Unique identifier for the run  [default:
                                  20241016_112218]
  --out-id, --out-id TEXT         Unique identifier for the run
  --in-prefix, --in-prefix TEXT   unique identifier for the run
  --write-csv / --no-write-csv    write CSV output at end of simulation
                                  [default: no-write-csv]
  --write-parquet / --no-write-parquet
                                  write parquet file output at end of
                                  simulation  [default: write-parquet]
  --help                          Show this message and exit.
```
to
```
twillard@epid-iss-MacBook-Pro ~/D/G/H/f/f/gempyor_pkg (multi-config)> flepimop simulate --help                                                            (/Users/twillard/Desktop/GitHub/HopkinsIDD/flepiMoP/venv)
Usage: flepimop simulate [OPTIONS] [CONFIG_FILES]...

      Forward simulate a model using gempyor.

      Args: (see auto generated CLI items below)

      Returns: None (side effect: writes output to disk)

  Command Line Interface arguments:

  config_files: Path

  config_filepath: Path

  seir_modifiers_scenarios: String

  outcome_modifiers_scenarios: String

  jobs: IntRange

  nslots: IntRange

  in_run_id: String

  out_run_id: String

  in_prefix: String

  first_sim_index: IntRange

  stoch_traj_flag: Bool

  write_csv: Bool

  write_parquet: Bool

Options:
  -c, --config PATH               Deprecated: configuration file for this
                                  simulation
  -s, --seir_modifiers_scenarios TEXT
                                  override/select the transmission scenario(s)
                                  to run
  -d, --outcome_modifiers_scenarios TEXT
                                  override/select the outcome scenario(s) to
                                  run
  -j, --jobs INTEGER RANGE        the parallelization factor  [default: 14;
                                  x>=1]
  -n, --nslots INTEGER RANGE      override the # of simulation runs in the
                                  config file  [x>=1]
  --in-id TEXT                    Unique identifier for the run
  --out-id TEXT                   Unique identifier for the run
  --in-prefix TEXT                unique identifier for the run
  -i, --first_sim_index INTEGER RANGE
                                  The index of the first simulation  [default:
                                  1; x>=1]
  --stochastic / --non-stochastic
                                  Run stochastic simulations?
  --write-csv / --no-write-csv    write csv output?  [default: no-write-csv]
  --write-parquet / --no-write-parquet
                                  write parquet output?  [default: write-
                                  parquet]
  --help                          Show this message and exit.
```
Is that the intent? Seems like some duplicated info.

Reply:
hmm, it is not; so this is a bit a click vs internal use conflict.

click uses the docstring on commands to generate its command line help. if we use the *same* function for direct calls, and want that to have generated docstring, that's going to spam-up the click help.

good catch - have to solve this with design differentiating the click command from the direct usage, which is something I think we want anyway?